LEGO1/mxdsobject.cpp

```diff
@@ -7,30 +7,123 @@
 MxDSObject::MxDSObject()
 {
   this->m_unk0c = 0;
-  this->m_unk10 = 0;
+  this->m_sourceName = NULL;
   this->m_unk14 = 0;
-  this->m_name = NULL;
+  this->m_objectName = NULL;
   this->m_unk24 = -1;
   this->m_unk1c = -1;
   this->m_unk28 = 0;
 }
 
+// OFFSET: LEGO1 0x100bf7e0
+MxDSObject::~MxDSObject()
+{
+  delete[] m_objectName;
+  delete[] m_sourceName;
+}
+
+// OFFSET: LEGO1 0x100bf870
+void MxDSObject::CopyFrom(MxDSObject &p_dsObject)
+{
+  this->SetSourceName(p_dsObject.m_sourceName);
+  this->m_unk14 = p_dsObject.m_unk14;
+  this->SetObjectName(p_dsObject.m_objectName);
+  this->m_unk1c = p_dsObject.m_unk1c;
+  this->m_unk24 = p_dsObject.m_unk24;
+  this->m_atomId = p_dsObject.m_atomId;
+  this->m_unk28 = p_dsObject.m_unk28;
+}
+
+// OFFSET: LEGO1 0x100bf8c0
+MxDSObject &MxDSObject::operator=(MxDSObject &p_dsObject)
+{
+  if (this == &p_dsObject)
+    return *this;
+
+  this->CopyFrom(p_dsObject);
+  return *this;
+}
+
 // OFFSET: LEGO1 0x100bf8e0
 void MxDSObject::SetObjectName(const char *p_name)
 {
-  if (p_name != this->m_name)
+  if (p_name != this->m_objectName)
   {
```

Comment:
Imho, only the curly braces of a method/function should be on a separate line.
```suggestion
  if (p_name != this->m_objectName) {
```